app/viewers/embed/viewer/common_viewer.rb

```diff
@@ -26,6 +26,20 @@ def external_url
         nil
       end
 
+      def any_stanford_only_files?
+        @purl_object.all_resource_files.any?(&:stanford_only?)
+      end
+
+      # indicates if viewer should display the Download All link in footer (override in specific viewer classes)
+      def display_download_all?
+        false
+      end
+
+      # link that will download all of the files in this object
+      def download_url
+        "#{Settings.stacks_url}/object/#{@purl_object.druid}"
+      end
+
       def stylesheet
```

Comment:
This moves some code from the File viewer (below) to the common viewer so it can be used by any viewer.  By default, the download all link is not shown, unless overridden by a specific content type viewer.